File: 0141-linked-list-cycle/0141-linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode:
#   def __init__(self, x):
#     self.val = x
#     self.next = None

class Solution:
  def hasCycle(self, head: Optional[ListNode]) -> bool:
    curr = head
    for _ in range(10001):
      if not curr: return False
      curr = curr.next
    return True

File: 0141-linked-list-cycle/0141-linked-list-cycle.java
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode next;
 *   ListNode(int x) {
 *     val = x;
 *     next = null;
 *   }
 * }
 */
public class Solution {
  public boolean hasCycle(ListNode head) {
    ListNode curr = head;
    for (int n = 0; n <= 10000; n++) {
      if (curr == null) return false;
      curr = curr.next;
    }
    return true;
  }
}